packages/cozy-device-helper/src/deeplink.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * This file is used to open the native app from a webapp
+ * if this native app is installed
+ *
+ * From a webapp, we don't have any clue if a native app is installed.
+ * The only way to know that, is to try to open the custom link
+ * (aka cozydrive://) and if nothing happens (no blur) we redirect to
+ * the callback
+ *
+ * Firefox tries to open custom link, so we need to create an iframe
+ * to detect if this is supported or not
+ */
+const _createHiddenIframe = (target, uri) => {
+  const iframe = document.createElement('iframe')
+  iframe.src = uri
+  iframe.id = 'hiddenIframe'
+  iframe.style.display = 'none'
+  target.appendChild(iframe)
+  return iframe
+}
+
+const openUriWithHiddenFrame = (uri, failCb) => {
+  const timeout = setTimeout(function() {
+    failCb()
+    handler && handler.remove()
+  }, 500)
+
+  let iframe = document.getElementById('hiddenIframe')
+  if (!iframe) {
+    iframe = _createHiddenIframe(document.body, 'about:blank')
+  }
+
+  var handler = window.addEventListener('blur', onBlur)
+  function onBlur() {
+    clearTimeout(timeout)
+    handler && handler.remove()
+  }
+
+  iframe.contentWindow.location.href = uri
+}
+
+const openUriWithTimeoutHack = (uri, failCb) => {
+  const timeout = setTimeout(function() {
+    failCb()
+    handler.remove()
+  }, 500)
+
+  //handle page running in an iframe (blur must be registered with top level window)
+  let target = window
+  while (target != target.parent) {
+    target = target.parent
+  }
+
+  var handler = target.addEventListener('blur', onBlur)
+  function onBlur() {
+    if (handler) {
+      clearTimeout(timeout)
+      handler.remove()
+    }
+  }
+
+  window.location = uri
+}
+
+const openUriWithMsLaunchUri = (uri, failCb) => {
+  navigator.msLaunchUri(uri, undefined, failCb)
+}
+
+const checkBrowser = () => {
+  const isOpera = !!window.opera || navigator.userAgent.indexOf(' OPR/') >= 0
+  const ua = navigator.userAgent.toLowerCase()
+  const isSafari =
+    (~ua.indexOf('safari') && !~ua.indexOf('chrome')) ||
```

Comment:
TIL https://www.joezimjs.com/javascript/great-mystery-of-the-tilde/ but uh… still a bit confused by this.